compiler/hash-typecheck/src/traverse.rs

```diff
@@ -976,7 +1117,14 @@ impl<'c, 'w, 'm, 'g, 'i> visitor::AstVisitor<'c> for ModuleTypechecker<'c, 'w, '
                     .iter()
                     .map(|&ty| match self.types().get(ty) {
                         TypeValue::Var(var) => Ok(*var),
-                        _ => Err(TypecheckError::BoundRequiresStrictlyTypeVars),
+                        _ => {
+                            // We need to get the location of the bound for error reporting
+                            let bound_location = node.bound.as_ref().unwrap().location();
```

Comment:
Yes because this error can only occur if a bound was present on struct definition right. Since the previous item maps `bound.as_ref().map(|bound| {...})` and thus we know that the bound exists here.